From 121184dbb3dae255d9a058446ed7d6750f8d6b91 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Fri, 21 Feb 2025 15:46:46 +0100
Subject: [PATCH 1627/1695] drivers: stpmic2: update the bindings

Update the bindings now that the interrupt framework is in use.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3d726498e377a24f97fdce01f757fe18e114e54f drivers: stpmic2: update the bindings]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I258766418bf51da33144ec99cee53e1dc2252cfe
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/441058
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 documentation/devicetree/bindings/mfd/st,stpmic2.yaml | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/documentation/devicetree/bindings/mfd/st,stpmic2.yaml b/documentation/devicetree/bindings/mfd/st,stpmic2.yaml
index a89dcbd54..b13668334 100644
--- a/documentation/devicetree/bindings/mfd/st,stpmic2.yaml
+++ b/documentation/devicetree/bindings/mfd/st,stpmic2.yaml
@@ -28,9 +28,8 @@ properties:
   reg:
     const: 0x33
 
-  st,wakeup-pin-number:
-    description: Pin used as wake-up source when for MP2
-    $ref: /schemas/types.yaml#/definitions/uint32
+  interrupts:
+    maxItems: 1
 
   st,pmic-it-id:
     description: PMIC interruptions ids to forward to the non-secure world
@@ -176,7 +175,7 @@ examples:
         compatible = "st,stpmic2";
         reg = <0x33>;
 
-        st,wakeup-pin-number = <1>;
+        interrupts-extended = <&exti1 52 IRQ_TYPE_EDGE_FALLING>;
         st,pmic-it-id = <IT_PKEY_FA IT_PKEY_RI>;
         st,notif-it-id = <0 1>;
 
-- 
2.47.2

